packages/channel-provider/src/types.ts

```diff
@@ -13,33 +13,44 @@ export type JsonRpcResponse<ResultType = any> = {
   result: ResultType;
 };
 
+export function isJsonRpcResponse(message: any): message is JsonRpcResponse {
+  return 'result' in message;
+}
+
 export type JsonRpcError = {
+  jsonrpc: '2.0';
   code: number;
   message: string;
   data?: {
     [key: string]: any;
   };
 };
 
+export type JsonRpcNotification = {
+  jsonrpc: '2.0';
+  method: string;
+  params: any;
+};
+
+export function isJsonRpcNotification(message: any): message is JsonRpcNotification {
+  return 'method' in message && !('id' in message);
+}
+
 export type JsonRpcErrorResponse = {
   id: number;
   jsonrpc: '2.0';
   error: JsonRpcError;
 };
 
+export function isJsonRpcErrorResponse(message: any): message is JsonRpcErrorResponse {
+  return 'error' in message;
+}
+
 export interface IChannelProvider {
   enable(url?: string): Promise<void>;
   send<ResultType = any>(method: string, params?: any[]): Promise<ResultType>;
```

Comment:
Did you forget to change this `any[]` to `any`?

Reply:
Yup, good catch!